plugin/pkg/auth/authenticator/token/keystone/keystone.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keystone
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"k8s.io/kubernetes/pkg/auth/user"
+	openstackcp "k8s.io/kubernetes/pkg/cloudprovider/providers/openstack"
```

Comment:
this is a tradeoff, I mean the share or not. In openstack cloud provider, there is not just keystone part, also the lb part, so if we share this, it does not looks good, as not all the information are needed here. also, let's say if there is rename in the openstack cloud provider(I mean the name in configuration file), it does not make sense to affect here, right? especially in case that the author does not know here at all.. so I would say it would be better to separate them, as easier to maintain.